server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -63,15 +63,58 @@
 import org.apache.james.mailbox.store.mail.model.impl.PropertyBuilder;
 import org.apache.james.mailbox.store.mail.model.impl.SimpleMailboxMessage;
 import org.apache.mailet.Mail;
-import org.javatuples.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
 import com.google.common.base.Splitter;
 import com.google.common.base.Throwables;
 
 public class SetMessagesCreationProcessor<Id extends MailboxId> implements SetMessagesProcessor<Id> {
+    
+    private static class Either<E, T> {
+
+        static <E, T> Either<E, T> value(T value) {
+            Preconditions.checkNotNull(value);
+            return new Either<>(null, value);
+        }
+        
+        static <E, T> Either<E, T> error(E error) {
+            Preconditions.checkNotNull(error);
+            return new Either<>(error, null);
+        }
+        
+        final E error;
+        final T value;
+        
+        Either(E error, T value) {
+            this.value = value;
+            this.error = error;
+        }
+
+        Either<E, T> validate(Function<T, Optional<E>> validator) {
```

Comment:
No, it adapts a validator (returning an optional error) to a Function that generates a Either